strategy.js

```diff
@@ -3,11 +3,11 @@ module.exports = function (passport, util) {
 
   function MockStrategy (options, verify) {
     if (!verify) { throw new TypeError('MockStrategy requires a verify callback'); }
-    if (!options.callbackURL) { throw new TypeError('MockStrategy requires a callbackURL'); }
+    if (!options.callbackURL && !options.client.redirect_uris[0]) { throw new TypeError('MockStrategy requires a callbackURL'); }
```

Comment:
I think this reads a little better to me. `if !(options.callbackURL || options.client.redirect_uris[0])`
nbd though